src/net_processing.cpp

```diff
@@ -2994,9 +2898,7 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         TxValidationState state;
 
         for (const GenTxid& gtxid : {GenTxid(false, txid), GenTxid(true, wtxid)}) {
-            nodestate->m_tx_download.m_tx_announced.erase(gtxid.GetHash());
-            nodestate->m_tx_download.m_tx_in_flight.erase(gtxid.GetHash());
-            EraseTxRequest(gtxid);
+            m_txrequest.ReceivedResponse(pfrom.GetId(), gtxid);
```

Comment:
The hash is different in both calls.

Reply:
Oops. Confused myself with the gtxid constructors. These are indeed different!

Thanks for removing the for loop. Makes it a lot clearer what's going on.

It would be better if these lines were above the `TxValidationState state;` declaration, but that's maybe unrelated to this PR, since the code you're replacing should also have been above that declaration.

Reply:
Yeah, trying to minimize diff - this shouldn't matter much.

Reply:
Agree it doesn't matter at all. It just offends my aesthetic sensibilities. Marking resolved.